app/uk/gov/hmrc/nrs/retrieval/controllers/NonrepRetrievalController.scala

```diff
@@ -19,61 +19,70 @@ package uk.gov.hmrc.nrs.retrieval.controllers
-                                         (implicit ec: ExecutionContext)
-  extends BackendController(controllerComponents) {
+class NonrepRetrievalController @Inject()(
+  val nonrepRetrievalConnector: NonrepRetrievalConnector,
+  override val controllerComponents: ControllerComponents)(implicit ec: ExecutionContext)
+    extends BackendController(controllerComponents) {
 
-   def search() = Action.async { implicit request =>
+  val search: Action[AnyContent] = Action.async { implicit request =>
     nonrepRetrievalConnector.search(mapToSeq(request.queryString)).map(response => Ok(response.body))
   }
 
-  def submitRetrievalRequest(vaultId: String, archiveId: String) = Action.async { implicit request =>
+  def submitRetrievalRequest(vaultId: String, archiveId: String): Action[AnyContent] = Action.async { implicit request =>
     nonrepRetrievalConnector.submitRetrievalRequest(vaultId, archiveId).map(response => rewriteResponse(response))
   }
 
-  def statusSubmissionBundle(vaultId: String, archiveId: String) = Action.async { implicit request =>
+  def statusSubmissionBundle(vaultId: String, archiveId: String): Action[AnyContent] = Action.async { implicit request =>
     nonrepRetrievalConnector.statusSubmissionBundle(vaultId, archiveId).map(response => rewriteResponse(response))
   }
 
-  def getSubmissionBundle(vaultId: String, archiveId: String) = Action.async { implicit request =>
-    nonrepRetrievalConnector.getSubmissionBundle(vaultId, archiveId)
-      .map{response => Ok(response.bodyAsBytes).withHeaders(mapToSeq(response.headers):_*)}
+  def getSubmissionBundle(vaultId: String, archiveId: String): Action[AnyContent] = Action.async { implicit request =>
+    nonrepRetrievalConnector.getSubmissionBundle(vaultId, archiveId).map { response =>
+      val errorMessage =
+        s"get submission bundle for vaultId: [$vaultId] archiveId: [$archiveId] received unexpected response status: ${response.status.toString}"
+
+      response.status match {
+        case NOT_FOUND                                 => throw new NotFoundException(errorMessage)
+        case status if status >= INTERNAL_SERVER_ERROR => throw new BadGatewayException(errorMessage)
+        case status if status >= MULTIPLE_CHOICES      => throw new InternalServerException(errorMessage)
+        case _                                         => Ok(response.bodyAsBytes).withHeaders(mapToSeq(response.headers): _*)
+      }
+    }
   }
 
-  private def rewriteResponse (response: HttpResponse) = {
+  private def rewriteResponse(response: HttpResponse) = {
     val headers: Seq[(String, String)] = mapToSeq(response.headers)
     response.status match {
-      case 200 => Ok(response.body).withHeaders(headers:_*)
-      case 202 => Accepted(response.body).withHeaders(headers:_*)
-      case 404 => NotFound(response.body)
-      case _ => Ok(response.body)
+      case OK        => Ok(response.body).withHeaders(headers: _*)
```

Comment:
IntelliJ was moaning about magic numbers

Reply:
Maybe this is the place to add some INFO logging to get some insights of what's transferred from NRS backend?